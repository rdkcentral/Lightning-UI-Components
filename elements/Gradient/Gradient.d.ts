import lng from '@lightningjs/core';
import Base from '../../Base';
import type { Color, StylePartial } from '../../types/lui';

export type GradientStyles = {
  gradientTop: string;
  gradientColor: Color;
  radius: lng.Tools.CornerRadius;
};

export default class Gradient extends Base {
  get style(): GradientStyles;
  set style(v: StylePartial<GradientStyles>);
}
