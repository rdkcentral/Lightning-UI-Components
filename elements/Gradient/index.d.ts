import Base from '../../Base';
import { WithThemeStylesConstructor } from '../../mixins/withThemeStyles';
import type { Tools } from '@lightningjs/core';

export interface GradientStyles {
  gradientTop?: string;
  gradientColor?: number;
  radius?: Tools.CornerRadius;
}

declare const Gradient_base: WithThemeStylesConstructor<
  typeof Base,
  GradientStyles
>;

export default class Gradient extends Gradient_base {
  style?: GradientStyles;
}
