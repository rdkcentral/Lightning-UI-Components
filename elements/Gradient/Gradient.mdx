import { Canvas, Story } from '@storybook/addon-docs';
import Gradient from '.';

# Gradient

A semitransparent Gradient to be used on top of any component, like Artwork.

## Source

https://github.comcast.com/Lightning/lightning-ui/packages/@lightning/ui-core/src/components/Gradient/Gradient.js

## Usage

Specify a width and height and overlay to Tiles or other items.

```js
import { Gradient } from '@lightning/ui';

class Example extends lng.Component {
  static _template() {
    return {
      Gradient: {
        type: Gradient,
        w: 320,
        h: 180
      }
    };
  }
}
```

<Canvas>
  <Story id="elements-gradient--basic" />
</Canvas>

## API

### Style Properties

| name          | type   | description                             |
| ------------- | ------ | --------------------------------------- |
| gradientColor | number | default starting color for the gradient |
| gradientTop   | number | default color the gradient fades to     |
| radius        | number | default gradient corner radius          |
