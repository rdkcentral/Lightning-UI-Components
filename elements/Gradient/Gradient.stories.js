import lng from '@lightningjs/core';

import Gradient from '.';
import mdx from './Gradient.mdx';
import kabob from '../../assets/images/kabob_320x180.jpg';

export default {
  title: 'elements / Gradient',
  parameters: {
    docs: {
      page: mdx
    }
  },
  args: {
    radius: 4,
    color: 'rgba(8,111,107,1)'
  },
  argTypes: {
    color: { name: 'Color', control: { type: 'color' } }
  }
};

export const Basic = args =>
  class Basic extends lng.Component {
    static _template() {
      return {
        Gradient: {
          type: Gradient,
          w: 320,
          h: 180,
          radius: args.radius,
          gradientColor: args.color
        }
      };
    }

    get _Gradient() {
      return this.tag('Gradient');
    }
  };
Basic.args = {};
Basic.argTypes = {};
Basic.parameters = {
  argActions: {}
};
