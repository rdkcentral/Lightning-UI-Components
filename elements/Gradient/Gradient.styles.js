export const base = theme => ({
  gradientTop: theme.color.transparent,
  radius: theme.radius.none
});

export const palette = theme => ({
  neutral: { gradientColor: theme.color.material },
  inverse: { gradientColor: theme.color.coreNeutral },
  brand: { gradientColor: theme.color.coreBrand }
});
