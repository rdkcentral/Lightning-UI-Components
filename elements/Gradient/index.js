import lng from '@lightningjs/core';
import Base from '../Base';
import { withStyles } from '../../mixins';
import { getValidColor } from '../../Styles/Styles';

export const styles = theme => ({
  radius: theme.border.radius.small,
  gradientColor: theme.palette.background.default,
  gradientTop: 'rgba(0,0,0,0)'
});

export default class Gradient extends withStyles(Base, styles) {
  static get properties() {
    return ['radius', 'gradientColor', 'gradientTop', 'radius'];
  }

  _construct() {
    super._construct();
    this._radius = this.styles.radius;
    this._gradientColor = getValidColor(this.styles.gradientColor);
    this._gradientTop = getValidColor(this.styles.gradientTop);
  }

  _update() {
    if (this.gradientColor) {
      this.patch({
        rect: true,
        rtt: true,
        colorTop: this.gradientTop,
        colorBottom: this.gradientColor,
        texture: lng.Tools.getRoundRect(this.w, this.h, this.radius)
      });
    }
  }

  _setGradientColor(color) {
    return getValidColor(color);
  }

  _setGradientTop(color) {
    return getValidColor(color);
  }
}
