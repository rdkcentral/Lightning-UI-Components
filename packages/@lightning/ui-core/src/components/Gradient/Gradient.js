import lng from '@lightningjs/core';
import Base from '../Base/index.js';
import * as styles from './Gradient.styles.js';

export default class Gradient extends Base {
  static get __componentName() {
    return 'Gradient';
  }

  static get __themeStyle() {
    return styles;
  }

  _update() {
    this.patch({
      rect: true,
      rtt: true,
      colorTop: this.style.gradientTop,
      colorBottom: this.style.gradientColor,
      texture: lng.Tools.getRoundRect(this.w, this.h, this.style.radius)
    });
  }
}
