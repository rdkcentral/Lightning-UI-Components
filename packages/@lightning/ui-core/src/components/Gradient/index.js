export { default as default } from './Gradient.js';
