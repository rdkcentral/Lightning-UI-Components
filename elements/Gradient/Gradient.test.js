import TestUtils from '../../test/lightning-test-utils';
import Gradient from '.';

const createComponent = TestUtils.makeCreateComponent(Gradient);

describe('Gradient', () => {
  let component, testRenderer;

  beforeEach(() => {
    [component, testRenderer] = createComponent();
  });
  afterEach(() => {
    component = null;
    testRenderer = null;
  });

  it('renders', () => {
    const tree = testRenderer.toJSON(2);
    expect(tree).toMatchSnapshot();
  });
});
