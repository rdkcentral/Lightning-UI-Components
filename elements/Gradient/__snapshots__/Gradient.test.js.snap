// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gradient renders 1`] = `
Object {
  "Component": Object {
    "active": false,
    "alpha": 1,
    "attached": true,
    "boundsMargin": null,
    "clipping": false,
    "color": 0,
    "colorBl": "ff141417",
    "colorBr": "ff141417",
    "colorUl": "0",
    "colorUr": "0",
    "enabled": true,
    "flex": false,
    "flexItem": false,
    "h": 0,
    "hasFinalFocus": true,
    "hasFocus": true,
    "isComponent": true,
    "mount": 0,
    "mountX": 0,
    "mountY": 0,
    "pivot": 0.5,
    "pivotX": 0.5,
    "pivotY": 0.5,
    "ref": "Component",
    "renderOfScreen": undefined,
    "renderToTexture": false,
    "scale": 1,
    "scaleX": 1,
    "scaleY": 1,
    "state": "",
    "tag": [Function],
    "tags": Array [
      "Component",
    ],
    "texture": Object {
      "type": "StaticCanvasTexture",
    },
    "type": "Gradient",
    "visible": true,
    "w": 0,
    "x": 0,
    "y": 0,
    "zIndex": 0,
  },
}
`;
